src/NHibernate/Type/OneToOneType.cs

```diff
@@ -92,6 +96,16 @@ public override ForeignKeyDirection ForeignKeyDirection
 
 		public override object Hydrate(DbDataReader rs, string[] names, ISessionImplementor session, object owner)
 		{
+			if (owner == null && names.Length > 0)
```

Comment:
This also fixes #2064. But #2082 still can be applied as it optimizes processing a bit by avoiding loaded entities "rehydrating".